x/concentrated-liquidity/tick.go

```diff
@@ -253,6 +253,68 @@ func (k Keeper) GetTickLiquidityForRange(ctx sdk.Context, poolId uint64) ([]quer
 	return liquidityDepthsForRange, nil
 }
 
+// GetLiquidityNetInDirection returns an array of liquidity depth in the given zeroForOne direction.
+// Uses boundTick if given, and iterates until we hit bound Tick's index. If not, uses min tick || max tick depending on the direction of zeroForOne.
+func (k Keeper) GetLiquidityNetInDirection(ctx sdk.Context, poolId uint64, zeroForOne bool, boundTickPointer *sdk.Int) ([]query.LiquidityDepth, error) {
```

Comment:
Why do we have to have `*sdk.Int`? Can we have `sdk.Int` and call `boundTickPointer.IsNil()` instead for simplicity?

Reply:
We're taking in a pointer from the request in the grpc router so we need somewhere to handle the pointer logic. Agreed that having pointers in arguments can cause anti-simplicity, moved the logic of handling this to grpc_query.go